### ISSUE TITLE
Vehicle Detection - train_test_split after normalizing features

### ISSUE THREAD

**Issue by mikkelkh** [udacity/sdc-issue-reports#490]
Throughout the Vehicle Detection and Tracking section, all code samples call the StandardScaler().fit() function before splitting datasets into training and test data.
This violates the general rule of not training any parameters on the test data.

Instead, train_test_split should be applied on the original images before normalizing, such that normalization parameters are only estimated on training data.

**Comment by andrewpaster** [comment]
Hey, yes, good point. Thanks for bringing this to our attention. Not sure yet when this will be corrected or if for the time being I'll just put a note in the classroom about it.

**Comment by mvirgo** [comment]
We have now updated these to avoid data snooping.
